pkg/interface/link/src/js/components/new.js

```diff
@@ -0,0 +1,239 @@
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      idName: '',
+      groups: [],
+      ships: [],
+      idError: false,
+      inviteError: false,
+      createGroup: true
+    };
+
+    this.titleChange = this.titleChange.bind(this);
+    this.descriptionChange = this.descriptionChange.bind(this);
+    this.setInvite = this.setInvite.bind(this);
+    this.createGroupChange = this.createGroupChange.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { props, state } = this;
+
+    if (prevProps !== props) {
+      let target = `/${state.idName}`;
+      if (target in props.resources) {
+        props.history.push(makeRoutePath(target));
+      }
+    }
+  }
+
+  titleChange(event) {
+    let asciiSafe = event.target.value.toLowerCase()
+      .replace(/[^a-z0-9-]/g, "-");
+    this.setState({
+      idName: asciiSafe + '-' + Math.floor(Math.random()*10000), // uniqueness
+      title: event.target.value
+    });
+  }
+
+  descriptionChange(event) {
+    this.setState({
+      description: event.target.value
+    });
+  }
+
+  setInvite(value) {
+    this.setState({
+      groups: value.groups,
+      ships: value.ships
+    });
+  }
+
+  createGroupChange(event) {
+    this.setState({
+      createGroup: !!event.target.checked,
+    });
+  }
+
+  onClickCreate() {
+    const { props, state } = this;
+
+    if (!state.title) {
+      this.setState({
+        idError: true,
+        inviteError: false
+      });
+      return;
+    }
+
+    let appPath = `/${state.idName}`;
+
+    if (appPath in props.resources) {
+      this.setState({
+        inviteError: false,
+        idError: true,
+        success: false
+      });
+      return;
+    }
+
+    let isValid = true;
+    let aud = state.ships.map(mem => `~${deSig(mem.trim())}`);
+    aud.forEach((mem) => {
+      if (!urbitOb.isValidPatp(mem)) {
+        isValid = false;
+      }
+    });
+
+    if (!isValid) {
+      this.setState({
+        inviteError: true,
+        idError: false,
+        success: false
+      });
+      return;
+    }
+
+    const target = state.groups.length === 0
+      ? {ships: aud}
+      : {group: state.groups[0]};
+
+    if (this.textarea) {
+      this.textarea.value = '';
+    }
+
+    this.setState({
+      error: false,
+      success: true,
+      group: [],
+      ships: []
+    }, () => {
+      api.setSpinner(true);
+      let submit = api.createCollection(
+        appPath,
+        state.title,
+        state.description,
+        target,
+        state.createGroup
+      );
+      submit.then(() => {
+        api.setSpinner(false);
+        props.history.push(makeRoutePath(appPath));
+      })
+    });
+  }
+
+  render() {
+    const { props, state } = this;
+
+    let createGroupClasses = state.createGroup
+      ? "relative checked bg-green2 br3 h1 toggle v-mid z-0"
+      : "relative bg-gray4 bg-gray1-d br3 h1 toggle v-mid z-0";
```

Comment:
I think we want `checked` and `toggle` classes from Chat's `custom.css` added to Link's `custom.css` too.